src/core/mac/mac.cpp

```diff
@@ -507,13 +509,65 @@ otError Mac::SetShortAddress(ShortAddress aShortAddress)
     return OT_ERROR_NONE;
 }
 
-otError Mac::SetChannel(uint8_t aChannel)
+otError Mac::SetPanChannel(uint8_t aChannel)
 {
     otError error = OT_ERROR_NONE;
 
     VerifyOrExit(OT_RADIO_CHANNEL_MIN <= aChannel && aChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_INVALID_ARGS);
 
-    mChannel = aChannel;
+    mPanChannel = aChannel;
+
+    VerifyOrExit(!mRadioChannelAcquisitionId);
+
+    mRadioChannel = mPanChannel;
+
+    UpdateIdleMode();
+
+exit:
+    return error;
+}
+
+otError Mac::SetRadioChannel(uint16_t aAcquisitionId, uint8_t aChannel)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(OT_RADIO_CHANNEL_MIN <= aChannel && aChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(mRadioChannelAcquisitionId && aAcquisitionId == mRadioChannelAcquisitionId,
+                 error = OT_ERROR_INVALID_STATE);
+
+    mRadioChannel = aChannel;
+
+    UpdateIdleMode();
+
+exit:
+    return error;
+}
+
+otError Mac::AcquireRadioChannel(uint16_t *aAcquisitionId)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aAcquisitionId != NULL, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(!mRadioChannelAcquisitionId, error = OT_ERROR_INVALID_STATE);
+
+    mRadioChannelAcquisitionId =
+        static_cast<uint16_t>(Random::GetUint32InRange(1, (1U << (sizeof(mRadioChannelAcquisitionId) * 8)) - 1));
+
+    *aAcquisitionId = mRadioChannelAcquisitionId;
+
+exit:
+    return error;
+}
+
+otError Mac::ReleaseRadioChannel(void)
```

Comment:
I am curious why not requiring acquisition id to be given as a parameter in release method?
With this we allow any user to release the radio channel? I guess there is a use-case for this?

Reply:
My thought on this is that I am not trying to prevent malicious behavior on the part of other openthread code.  I am merely providing a mechanism for the code that owns the lock to release that lock.  I don't care to prevent other code from unlocking it without first owning it, because that would be a bug and we don't write buggy code.